src/main/java/apoc/periodic/BatchMode.java

```diff
@@ -0,0 +1,44 @@
+package apoc.periodic;
+
+import apoc.util.Util;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+enum BatchMode {
+    BATCH,
+    BATCH_SINGLE,
+    SINGLE;
+
+
+    private static final Map<String, BatchMode> nameIndex =  new HashMap<>();
+    static {
+        for (BatchMode batchMode : BatchMode.values()) {
+            nameIndex.put(batchMode.name(), batchMode);
+        }
+    }
+
+   static BatchMode fromIterateList(boolean iterateList) {
+        return iterateList ? BATCH : SINGLE;
+    }
+
+    static BatchMode fromConfig(Map<String,Object> config) {
+        Object batchMode = config.get("batchMode");
+
+        if(batchMode != null) {
+            BatchMode lookedUpBatchMode = nameIndex.get(batchMode.toString().toUpperCase());
```

Comment:
if we get rid of `nameIndex` you can change it into:
```suggestion
            BatchMode lookedUpBatchMode = BatchMode.valueOf(batchMode.toString().toUpperCase());
```

Reply:
And we can also get rid of the following exception